### ISSUE TITLE
fetch_url always caches resolved URLs

### ISSUE THREAD

**Issue by chreke** [jfinkhaeuser/prance#88]
The `util.url.fetch_url` and `util.url.fetch_url_text` functions are supposed to have an optional cache ("Mapping cache: An optional cache. If the URL can be found in the cache, return the cache contents.", as per the documentation)

However, in the current implementation there will *always* be a cache, which I believe was not the intention.

The root cause is using `{}` as default for the `cache` argument; rather than re-instantiating a new dictionary every time the function is called, the same dictionary will be re-used. See: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

A simple fix would be:

```python
def fetch_url(url, cache = None, encoding = None):
  cache = cache or {}
```

## Expected Behaviour

URL contents should not be cached unless the `cache` argument is provided.

## Actual Behaviour

URL contents get cached if no `cache` argument is provided.

## Environment
- OS: MacOS 10.15.7
- Python version: 3.8
- Swagger/OpenAPI version used: 3.02
- Backend: openapi-spec-validator

@jfinkhaeuser

**Comment by jfinkhaeuser** [comment]
You're absolutely right! That's one of the things in Python I keep forgetting about. There will be similar issues; I'll go through the code to find them.
